api/src/auth/drivers/oauth2.ts

```diff
@@ -125,7 +125,7 @@ export class OAuth2AuthDriver extends LocalAuthDriver {
 
 		const email = userInfo[emailKey ?? 'email'] as string | null | undefined;
 		// Fallback to email if explicit identifier not found
-		const identifier = (userInfo[identifierKey] as string | null | undefined) ?? email;
+		const identifier = (userInfo[identifierKey] as any | null | undefined)?.toString() ?? email;
```

Comment:
Should do this, just to handle the edge case if the email is not a string. 🤞 
```suggestion
		const identifier = (userInfo[identifierKey] as any | null | undefined ?? email)?.toString();
```